src/borrow_tracker/tree_borrows/tree.rs

```diff
@@ -128,6 +128,22 @@ impl LocationState {
         Ok(transition)
     }
 
+    /// Like `perform_access`, but ignores the diagnostics, and also is pure.
```

Comment:
```suggestion
    /// Like `perform_access`, but ignores the concrete error cause and also uses state-passing
    /// rather than a mutable reference.
```